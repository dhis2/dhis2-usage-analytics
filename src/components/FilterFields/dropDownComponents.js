import React from 'react'
import i18n from '@dhis2/d2-i18n'
import { SelectField } from '@dhis2/ui-core'
import withMargin from './withMargin'

import CATEGORIES from '../../constants/categories'
import INTERVALS from '../../constants/intervals'
import AGGREGATIONS from '../../constants/aggregations'
import CHART_TYPES from '../../constants/chartTypes'
import EVENT_TYPES from '../../constants/eventTypes'
import PAGE_SIZES from '../../constants/pageSizes'
import SORT_ORDERS from '../../constants/sortOrders'

const FIELD_KIND = 'filled'

const withKind = props => ({ ...props, [FIELD_KIND]: true })
const SelectFieldWithMargin = withMargin(SelectField)

export const Category = props => (
    <SelectFieldWithMargin
        {...withKind(props)}
        name="category"
        label={i18n.t('Category')}
        kind={FIELD_KIND}
    >
        {CATEGORIES.map(i => (
            <option value={i.value}>{i.label}</option>
        ))}
    </SelectFieldWithMargin>
)

export const Interval = props => (
    <SelectFieldWithMargin
        {...withKind(props)}
        name="interval"
        label={i18n.t('Interval')}
        kind={FIELD_KIND}
    >
        {INTERVALS.map(i => (
            <option value={i.value}>{i.label}</option>
        ))}
    </SelectFieldWithMargin>
)

export const AggregationLevel = props => (
    <SelectFieldWithMargin
        {...withKind(props)}
        name="aggregationLevel"
        label={i18n.t('Aggregation Level')}
        kind={FIELD_KIND}
    >
        {AGGREGATIONS.map(i => (
            <option value={i.value}>{i.label}</option>
        ))}
    </SelectFieldWithMargin>
)

export const ChartType = props => (
    <SelectFieldWithMargin
        {...withKind(props)}
        name="chartType"
        label={i18n.t('Chart Type')}
        kind={FIELD_KIND}
    >
        {CHART_TYPES.map(i => (
            <option value={i.value}>{i.label}</option>
        ))}
    </SelectFieldWithMargin>
)

export const EventType = props => (
    <SelectFieldWithMargin
        {...withKind(props)}
        name="eventType"
        label={i18n.t('Event Type')}
        kind={FIELD_KIND}
    >
        {EVENT_TYPES.map(i => (
            <option value={i.value}>{i.label}</option>
        ))}
    </SelectFieldWithMargin>
)

export const PageSize = props => (
    <SelectFieldWithMargin
        {...withKind(props)}
        name="pageSize"
        label={i18n.t('Page Size')}
        kind={FIELD_KIND}
    >
        {PAGE_SIZES.map(i => (
            <option value={i.value}>{i.label}</option>
        ))}
    </SelectFieldWithMargin>
)

export const SortOrder = props => (
    <SelectFieldWithMargin
        {...withKind(props)}
        name="sortOrder"
        label={i18n.t('Sort Order')}
        kind={FIELD_KIND}
    >
        {SORT_ORDERS.map(i => (
            <option value={i.value}>{i.label}</option>
        ))}
    </SelectFieldWithMargin>
)
