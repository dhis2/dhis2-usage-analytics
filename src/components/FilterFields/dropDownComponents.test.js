import React from 'react'
import { shallow } from 'enzyme'
import * as DropDowns from './dropDownComponents'

const onChange = jest.fn()

const defaultProps = {
    onChange,
    value: 'test',
}

describe('dropDownComponents', () => {
    it('<Category/> matches the snapshot', () => {
        const wrapper = shallow(<DropDowns.Category {...defaultProps} />)
        expect(wrapper).toMatchSnapshot()
    })
    it('<Interval/> matches the snapshot', () => {
        const wrapper = shallow(<DropDowns.Interval {...defaultProps} />)
        expect(wrapper).toMatchSnapshot()
    })
    it('<AggregationLevel/> matches the snapshot', () => {
        const wrapper = shallow(
            <DropDowns.AggregationLevel {...defaultProps} />
        )
        expect(wrapper).toMatchSnapshot()
    })
    it('<ChartType/> matches the snapshot', () => {
        const wrapper = shallow(<DropDowns.ChartType {...defaultProps} />)
        expect(wrapper).toMatchSnapshot()
    })
    it('<EventType/> matches the snapshot', () => {
        const wrapper = shallow(<DropDowns.EventType {...defaultProps} />)
        expect(wrapper).toMatchSnapshot()
    })
    it('<PageSize/> matches the snapshot', () => {
        const wrapper = shallow(<DropDowns.PageSize {...defaultProps} />)
        expect(wrapper).toMatchSnapshot()
    })
    it('<SortOrder/> matches the snapshot', () => {
        const wrapper = shallow(<DropDowns.SortOrder {...defaultProps} />)
        expect(wrapper).toMatchSnapshot()
    })
})
