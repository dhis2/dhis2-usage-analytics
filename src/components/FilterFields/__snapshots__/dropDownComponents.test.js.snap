// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dropDownComponents <AggregationLevel/> matches the snapshot 1`] = `
<ComponentWithMargin
  filled={true}
  label="Aggregation Level"
  name="aggregationLevel"
  onChange={[MockFunction]}
  value="test"
>
  <option
    value="SUM"
  >
    Sum
  </option>
  <option
    value="AVERAGE"
  >
    Average
  </option>
</ComponentWithMargin>
`;

exports[`dropDownComponents <Category/> matches the snapshot 1`] = `
<ComponentWithMargin
  filled={true}
  label="Category"
  name="category"
  onChange={[MockFunction]}
  value="test"
>
  <option
    value="FAVORITE_VIEWS"
  >
    Favorite views
  </option>
  <option
    value="FAVORITES_SAVED"
  >
    Favorites saved
  </option>
  <option
    value="USERS"
  >
    Users
  </option>
  <option
    value="TOP_FAVORITES"
  >
    Top favorites
  </option>
  <option
    value="DATA_VALUES"
  >
    Data values
  </option>
</ComponentWithMargin>
`;

exports[`dropDownComponents <ChartType/> matches the snapshot 1`] = `
<ComponentWithMargin
  filled={true}
  label="Chart Type"
  name="chartType"
  onChange={[MockFunction]}
  value="test"
>
  <option
    value="ALL"
  >
    All
  </option>
  <option
    value="TOTAL"
  >
    Total
  </option>
</ComponentWithMargin>
`;

exports[`dropDownComponents <EventType/> matches the snapshot 1`] = `
<ComponentWithMargin
  filled={true}
  label="Event Type"
  name="eventType"
  onChange={[MockFunction]}
  value="test"
>
  <option
    value="CHART_VIEW"
  >
    Chart
  </option>
  <option
    value="MAP_VIEW"
  >
    Map
  </option>
  <option
    value="REPORT_TABLE_VIEW"
  >
    Report table
  </option>
  <option
    value="EVENT_REPORT_VIEW"
  >
    Event report
  </option>
  <option
    value="EVENT_CHART_VIEW"
  >
    Event chart
  </option>
  <option
    value="DASHBOARD_VIEW"
  >
    Dashboard
  </option>
  <option
    value="DATA_SET_REPORT_VIEW"
  >
    Data set report
  </option>
</ComponentWithMargin>
`;

exports[`dropDownComponents <Interval/> matches the snapshot 1`] = `
<ComponentWithMargin
  filled={true}
  label="Interval"
  name="interval"
  onChange={[MockFunction]}
  value="test"
>
  <option
    value="DAY"
  >
    Day
  </option>
  <option
    value="WEEK"
  >
    Week
  </option>
  <option
    value="MONTH"
  >
    Month
  </option>
  <option
    value="YEAR"
  >
    Year
  </option>
</ComponentWithMargin>
`;

exports[`dropDownComponents <PageSize/> matches the snapshot 1`] = `
<ComponentWithMargin
  filled={true}
  label="Page Size"
  name="pageSize"
  onChange={[MockFunction]}
  value="test"
>
  <option
    value={5}
  >
    5
  </option>
  <option
    value={10}
  >
    10
  </option>
  <option
    value={15}
  >
    15
  </option>
  <option
    value={20}
  >
    20
  </option>
  <option
    value={25}
  >
    25
  </option>
</ComponentWithMargin>
`;

exports[`dropDownComponents <SortOrder/> matches the snapshot 1`] = `
<ComponentWithMargin
  filled={true}
  label="Sort Order"
  name="sortOrder"
  onChange={[MockFunction]}
  value="test"
>
  <option
    value="ASC"
  >
    Ascending
  </option>
  <option
    value="DESC"
  >
    Descending
  </option>
</ComponentWithMargin>
`;
